src/main/scala/com/exasol/cloudetl/kafka/KafkaTopicDataImporter.scala

```diff
@@ -31,75 +25,21 @@ object KafkaTopicDataImporter extends LazyLogging {
   def run(metadata: ExaMetadata, iterator: ExaIterator): Unit = {
     val kafkaProperties = KafkaConsumerProperties(iterator.getString(0), metadata)
     val partitionId = iterator.getInteger(1)
-    val partitionOffset = iterator.getLong(2)
-    val partitionNextOffset = partitionOffset + 1L
+    val partitionNextOffset = iterator.getLong(2) + 1L
     val nodeId = metadata.getNodeId()
     val vmId = metadata.getVmId()
     logger.info(
       s"Starting Kafka consumer for partition '$partitionId' at next offset " +
         s"'$partitionNextOffset' for node '$nodeId' and vm '$vmId'."
     )
-
-    val topic = kafkaProperties.getTopic()
-    val topicPartition = new TopicPartition(topic, partitionId)
-
-    val fieldSpecs = FieldParser.get(kafkaProperties.getRecordFields())
-    val recordDeserializers = DeserializationFactory.getSerializers(fieldSpecs, kafkaProperties)
-    val kafkaConsumer = KafkaConsumerFactory(
+    val kafkaRecordConsumer = KafkaRecordConsumer(
       kafkaProperties,
-      recordDeserializers.keyDeserializer,
-      recordDeserializers.valueDeserializer
+      partitionId,
+      partitionNextOffset,
+      metadata.getOutputColumnCount().toInt,
```

Comment:
Maybe also extract a variable to be consistent?

Reply:
Fixed.